handbook/product/rollout_process.md

```diff
@@ -0,0 +1,44 @@
+# Feature rollout process
+
+Features come in many different sizes and shapes, and the process for introducing new functionality ranges with these differences. For large or significantly impactful changes or changes that simply need a bit more time to bake, it is encouraged that the following rollout process is followed.
+
+## Sourcegraph Cloud
+
+Sourcegraph Cloud is continuously deployed with all new updates to master. We maintain a [releasability contract](../engineering/continuous_releasability.md) and require all new features to be released behind a feature flag to ensure that functionality can be turned off if a problem arises.
```

Comment:
I don't think it's good to reference the releasability contract here. For example, [private code CTAs](https://github.com/sourcegraph/sourcegraph/pull/12288#discussion_r456694318) shouldn't be feature flagged, IMO, but it seems that product does want to discuss/approve this before merging (https://sourcegraph.slack.com/archives/C0C324C91/p1595009322049800). Should we have two different paths for features that should and shouldn't be feature flagged?

Reply:
@attfarhan, curious why think it shouldn't be feature flagged. Is it because it's primarily a UI change that can be fixed easily (instead of reverted?). I can add exceptions we decide on to the releasability contract in a follow-up PR for feedback.
Private code CTA was behind a master feature (`showRepogroupHomepage`) with one non-ideal way to turn it off if a problem arises. To your point, we don't have a section for how incremental changes should be rolled out. Would that be helpful in this doc?

Reply:
Yeah I'm basing it off the fact that we _didn't_ feature flag it. It was an incremental change, as you said. But yeah you're right that it was behind an umbrella feature flag. I think guidelines for incremental changes could be helpful.